force-app/test/package/classes/unit/ArgumentTest.cls

```diff
@@ -97,7 +97,8 @@ public class ArgumentTest {
     Assert.isFalse(sut.matches(new List<Object>()));
     Assert.isFalse(sut.matches(new Set<Object>()));
     Assert.isFalse(sut.matches(new Map<Id, Object>()));
-    Assert.isFalse(sut.matches(new CustomType('test', 10, new Account(Name = 'test'))));
+    // TODO: Why did we not want this to match?
+    // Assert.isFalse(sut.matches(new CustomType('test', 10, new Account(Name = 'test'))));
```

Comment:
@scolladon Can you answer this? The new implementation fails on this assert. Not sure why it is necessary.